### DISCUSSION TITLE
Getting "Duplicate name, rejecting enrollment" after wazuh agents upgrade

### DISCUSSION THREAD

**Discussion by TrailByte** [wazuh/wazuh#26364] [Q&A] [answered]
Hello all! 
I am kinda new to wazuh. Recently I upgraded multiple agents from some very old versions (4.1.0-4.6.0) to the latest (4.9.0). After that i started getting the "WARNING: Duplicate name '<name>', rejecting enrollment. Agent '<id>' can't be replaced since it is not disconnected." warning for all of them. I saw in the agents ossec.conf that the enrollment is enabled. Could the new agent version be trying to re-enroll itself because of it? 

If that is the case, what can I do to solve it?

**Comment by MiguelazoDS** [comment, marked as answer]
Hello @MikeKall, 

Yes, it could be the reason, they were too old. 

The error you're seeing is related to the force block option for authd 

https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/auth.html#force

Basically, it is trying to replace the old agent with a new one, but since the agent is not disconnected, it is preventing it from doing so. 

You can edit the force options, or re-register your agent (removing and registering once again) .

**Reply by TrailByte** [reply]
Hello @MiguelazoDS. Thanks for the info, I was in the middle of configuring the manager to force replace the old agents when I took a better look in the manager logs and saw this: 
```
wazuh-authd: INFO: Received request for a new agent (<hostname>) from: **127.0.0.1**
wazuh-authd: WARNING: Duplicate name '<hostname>', rejecting enrollment. Agent '12710' can't be replaced since it is not disconnected.

```

The thing that is troubling me is that the new agent is coming from the localhost IP. I am not sure this is supposed to happen. This is the configuration from the agent in case there is a misconfiguration that I don't get:

```
  <client>
    <server>
      <address>10.60.9.32</address>
      <port>1514</port>
      <protocol>tcp</protocol>
      <max_retries>5</max_retries>
      <retry_interval>5</retry_interval>
    </server>
    <notify_time>10</notify_time>
    <time-reconnect>60</time-reconnect>
    <auto_restart>yes</auto_restart>
    <crypto_method>aes</crypto_method>

    <enrollment>
      <enabled>yes</enabled>
      <manager_address>10.60.9.32</manager_address>
      <agent_name><hostname></agent_name>
      <port>1515</port>
      <authorization_pass_path>/var/ossec/etc/authd.pass</authorization_pass_path>
      <auto_method>no</auto_method>
      <delay_after_enrollment>20</delay_after_enrollment>
      <use_source_ip>no</use_source_ip>
    </enrollment>

  </client>
```

**Comment by MiguelazoDS** [comment]
Hi @MikeKall 

That's strange, are you running containers? is the agent IP the same as the manager? How is your environment configured? 
I'm wondering if that is possible to happen with containers.

**Reply by TrailByte** [reply]
Hello @MiguelazoDS, all the wazuh components run on Red Hat 9.4 VMs and none of the agents have the same IP as the master. The environment is very simple as of now, all the agents talk directly to the master. I am planning on deploying a load balancer and 2 workers since the master seems to have troubles processing all the requests from 2000+ agents. 
But for now, do you may ave any tips on how can I troubleshoot the localhost IP thing? It seems very weird to be honest.

**Reply by TrailByte** [reply]
Ok, I found the "problem". It's the way the agents communicate with the master. It's not a problem any more. Thank you fore your help @MiguelazoDS, appreciate it!

**Reply by MiguelazoDS** [reply]
I'm glad to know you could solve it! Regards!
